### PR TITLE
Update DocumentationMode docs

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes https://github.com/dotnet/roslyn-api-docs/issues/86

### CODE REVIEW

File: src/Compilers/Core/Portable/Compilation/ParseOptions.cs [hunk 1]
```diff
@@ -32,9 +32,13 @@ public abstract class ParseOptions
         public SourceCodeKind SpecifiedKind { get; protected set; }
 
         /// <summary>
-        /// Gets a value indicating whether the documentation comments are parsed.
+        /// Gets a value indicating whether the documentation comments are parsed and can produce diagnostics, parsed only, or treated as regular comments.
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Compilation/ParseOptions.cs:35]: Consider simply saying: "Gets a value indicating whether the documentation comments are parsed and analyzed."

File: src/Compilers/Core/Portable/Compilation/ParseOptions.cs [hunk 2]
```diff
@@ -32,9 +32,13 @@ public abstract class ParseOptions
...
         /// </summary>
-        /// <value><c>true</c> if documentation comments are parsed, <c>false</c> otherwise.</value>
+        /// <value>
+        /// <para><see cref="DocumentationMode.Diagnose"/> if documentation comments are parsed and can produce diagnostics.</para>
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Compilation/ParseOptions.cs:38]: Consider saying: "if documentation comments are parsed and analyzed"

File: src/Compilers/Core/Portable/Compilation/ParseOptions.cs [hunk 3]
```diff
@@ -32,9 +32,13 @@ public abstract class ParseOptions
...
+        /// <para><see cref="DocumentationMode.Parse"/> if documentation comments are parsed without producing diagnostics.</para>
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Compilation/ParseOptions.cs:39]: Consider saying: "if documentation comments are parsed, but not analyzed." I am assuming parsing itself can cause some diagnostics to be reported. So, "without producing diagnostics" is probably not accurate.

**Comment by AlekseyTs** [comment]
Done with review pass (commit 1)

File: src/Compilers/Core/Portable/Compilation/ParseOptions.cs (hunk 3 above)
Comment (Youssef1313) [reply src/Compilers/Core/Portable/Compilation/ParseOptions.cs:39]: The doc for `DocumentationMode.Parse` says: "Parses documentation comments as structured trivia, but do not report any diagnostics.". I'll try to check if parsing itself can produce diagnostics or not, and update both comments.

Comment (AlekseyTs) [reply src/Compilers/Core/Portable/Compilation/ParseOptions.cs:39]: If enum values are documented, then why duplicate the information here at all? This is going to create an unnecessary maintenance burden, in my opinion.

Comment (Youssef1313) [reply src/Compilers/Core/Portable/Compilation/ParseOptions.cs:39]: Sounds reasonable. I'm going to delete the "value" tag.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 2)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For a second review.

**Review by 333fred — APPROVED** [review]

**Comment by 333fred** [comment]
Thanks @Youssef1313!

**Pull request merged**